impala/dbapi/interface.py

```diff
@@ -175,4 +237,17 @@ def _bind_parameters(operation, parameters):
             string_parameters[name] = "'" + _escape(value) + "'"
         else:
             string_parameters[name] = str(value)
+
+    # replace named parameters by their pyformat equivalents
+    operation = re.sub(":([a-zA-Z]+)", "%(\g<1>)s", operation)
```

Comment:
I don't know whether there is a "spec" for the "named" mechanism, but can the identifiers include digits and underscores?  cf. http://stackoverflow.com/a/10134719/510187